### ISSUE TITLE
Suggest box_string

### ISSUE THREAD

**Issue by lengyijun** [rust-lang/rust-clippy#7451]
### What it does

Check for usage for `Box<String>` in struct definition and function parameter.

### Categories (optional)

Perf, Warn

*What is the advantage of the recommended code over the original code*

`String` already allocates on heap, `Box<String>` allocates a pointer on heap.
Similar lints:  `rc_buffer`, `box_vec`, `redundant_allocation`,

### Drawbacks

None.

### Example

```rust
struct foo(Box<String>)
fn bar(a: Box<String>){}
```

Could be written as:

```rust
struct foo(String)
fn bar(a: String){}
```

**Comment by camsteffen** [comment]
I think we should enhance and rename `box_vec`.
